examples/workbooks/Products&Reviews.ts

```diff
@@ -0,0 +1,527 @@
+                "validate",
+            );
+        }
+
+        return;
+    };
+
+const validatePositive = (value: number) => (): void | FF.Message => {
+    if (value < 0.01) {
+        return new FF.Message(
+            "Value must be at least 1 cent.",
+            "error",
+            "validate",
+        );
+    }
+
+    return;
+};
+
+const validateUpcOrEan = (value: string) => (): void | FF.Message => {
+    if (isFalsy(/\d{12,13}/g.test(value))) {
+        return new FF.Message("Must be 12 or 13 digits.", "error", "validate");
+    }
+
+    // any 12-digit UPC can be converted into a 13-digit EAN
+    const padded: string = value.padStart(13, "0");
+
+    const allDigits: ReadonlyArray<number> = padded.split("").map(Number);
+    const checksumDigit: number = (allDigits as Array<number>).pop() ?? 0;
+
+    const evenDigits: ReadonlyArray<number> = allDigits.filter(
+        (_value, index) => index % 2 === 0,
+    );
+    const oddDigits: ReadonlyArray<number> = allDigits.filter(
+        (_value, index) => index % 2 !== 0,
+    );
+    const sumEvens: number = evenDigits.reduce((acc, digit) => acc + digit, 0);
+    const sumOdds: number = oddDigits.reduce((acc, digit) => acc + digit, 0);
+    const remainder: number = (sumOdds * 3 + sumEvens) % 10;
+
+    if (10 - remainder !== checksumDigit) {
+        return new FF.Message("Invalid UPC or EAN.", "error", "validate");
+    }
+
+    return;
+};
+
+/*
+ * Record Hooks
+ */
+
+const defaultUUID =
+    (key: string) =>
+        (record: FlatfileRecord): FlatfileRecord => {
+            const value = record.get(key);
+
+            if (isNil(value)) {
+                record
+                    .set(key, uuidv4())
+                    .addComment(key, "Value was autogenerated for you.");
+            }
+
+            return record;
+        };
+
+/*
+ * Helpers
+ */
+
+/**
+ * Easily pass the result of one function as the input of another.
+ *
+ * @example
+ * pipe(fn1, fn2, ...);
+ */
+const pipe = (...fns: Array<any>) => fns.reduce((acc, fn) => fn(acc));
+
+/**
+ * Converts `String.prototype.trim()` to a normal fn so it can be used with `pipe`.
+ *
+ * @param {string} value - value to apply operation.
+ *
+ * @example
+ * pipe(value, trim);
+ */
+const trim = (value: string): string => value.trim();
+
+/**
+ * Allows us to combine multiple validations in a quick and easy way.
+ *
+ * @example
+ * runValidations(fn1, fn2, fn3, ...);
+ */
+const runValidations = (...fns: Array<any>): Array<FF.Message> => {
+    return fns.reduce((acc, fn) => [...acc, fn()], []).filter(isNotNil);
+};
+
+/**
+ * Allows us to sequence multiple RecordHooks _synchronously_ on a `FlatfileRecord`.
+ *
+ * @example
+ * runRecordHooks(fn1, fn2, fn3, ...)(record)
+ */
+export const runRecordHooks =
+    (...fns: Array<(x: FlatfileRecord) => void>) =>
+        (x: FlatfileRecord) =>
+            fns.map((f) => f(x));
+
+/*
+ * Main
+ */
+
+const ReviewsSheet = new FF.Sheet(
+    "Reviews",
```

Comment:
I would expect that the Reviews sheet would have a LinkedField referencing product

Reply:
We discussed not making this a `LinkedField` example the other day. That will go in another, future, example.